pkg/metrics/garden.go

```diff
@@ -0,0 +1,397 @@
+	c.collectProjectCountMetrics(ch)
+	c.collectUserCountMetric(ch)
+	c.collectShootMetrics(ch)
+}
+
+// collectShootMetrics collect Shoot metrics, which contain information to Shoot itself, their state and usage.
+func (c gardenMetricsCollector) collectShootMetrics(ch chan<- prometheus.Metric) {
+	var (
+		nodeCountTotal float64
+
+		usageRegionCounter  = make(map[string]float64)
+		usageSeedCounter    = make(map[string]float64)
+		usageVersionCounter = make(map[string]float64)
+
+		operationCounter = make(map[string]float64)
+		regionCounter    = make(map[string]float64)
+		seedCounter      = make(map[string]float64)
+		stateCounter     = make(map[string]float64)
+		versionCounter   = make(map[string]float64)
+	)
+
+	// Set operation and state counter to 0
+	for _, cp := range cloudProviders {
+		for _, state := range shootStates {
+			stateCounter[fmt.Sprintf("%s:%s", cp, state)] = 0
+		}
+		for _, operation := range shootOperations {
+			operationCounter[fmt.Sprintf("%s:%s", cp, operation)] = 0
+		}
+	}
+
+	// Set for all available lists (regions, seeds and version) the corrsponding counters to 0.
+	// Each element in the counter maps will be increased, if a Shoot match to the element during
+	// the processing of the Shoots below. The first run will be ignored, because the list get
+	// initially poupulated during the first metrics collection run.
+	if !firstRun {
+		for _, p := range shootPhases {
+			for _, r := range regionList {
+				regionCounter[fmt.Sprintf("%s:%s", r, p)] = 0
+			}
+			for _, s := range seedList {
+				seedCounter[fmt.Sprintf("%s:%s", s, p)] = 0
+			}
+			for _, v := range versionList {
+				versionCounter[fmt.Sprintf("%s:%s", v, p)] = 0
+			}
+		}
+	}
+
+	// Fetch all Shoots.
+	shoots, err := c.k8sGardenClient.GardenClientset().GardenV1beta1().Shoots(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		ScrapeFailures.With(prometheus.Labels{"kind": "shoots"}).Inc()
+		return
+	}
+
+	for _, shoot := range shoots.Items {
+		var (
+			isSeed    bool
+			mailTo    string
+			nodeCount int
+		)
+
+		// Sanity check: Seed specified
+		if shoot.Spec.Cloud.Seed == nil {
+			continue
+		}
+		seed := *(shoot.Spec.Cloud.Seed)
+
+		// Determine the cloud provider.
+		cloudProvider, err := helper.DetermineCloudProviderInShoot(shoot.Spec.Cloud)
+		if err != nil {
+			logger.Logger.Infof("Unable to determine cloud provider for Shoot %s", shoot.Name)
+			return
+		}
+		var (
+			iaas        = string(cloudProvider)
+			versionIaas = fmt.Sprintf("%s:%s", iaas, shoot.Spec.Kubernetes.Version)
+			seedIaas    = fmt.Sprintf("%s:%s", iaas, seed)
+			regionIaas  = fmt.Sprintf("%s:%s", iaas, shoot.Spec.Cloud.Region)
+		)
+
+		// Check if the Shoot is a Seed and get the mail of the operator/owner if available.
+		if shoot.Annotations != nil {
+			mailTo = shoot.Annotations[common.GardenOperatedBy]
+
+			ok, err := strconv.ParseBool(shoot.Annotations[common.ShootUseAsSeed])
```

Comment:
You can use `helper.IsUsedAsSeed(shoot)` of the `pkg/apis/garden/v1beta1/helper` package.